backend/services/job.py

```
@@ -53,31 +45,109 @@ async def schedule_agent_job(payload:RunAgentRequest):
+        if current_status:
+            await redisservices.set_job_status(job_id, current_status.model_dump(mode="json"))
+        raise
+
     return job_id
 
 def update_job_status(job_id: str, update: Dict[Any, Any]) -> bool:
     if job_id not in job_results:
+        if "created_at" in update:
+            try:
+                job_results[job_id] = JobStatusResponse(**update)
+            except Exception:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)